dynamiqs/utils/utils.py

```diff
@@ -34,6 +36,42 @@ def is_ket(x: Tensor) -> bool:
     return x.size(-1) == 1
 
 
+def is_bra(x: Tensor) -> bool:
+    """Returns True if a tensor is in the format of a bra.
+
+    Args:
+        x (...): Tensor.
+
+    Returns:
+        True if the second to last dimension of `x` is 1, False otherwise.
+    """
+    return x.size(-2) == 1
+
+
+def is_dm(x: Tensor) -> bool:
+    """Returns True if a tensor is in the format of a density matrix.
+
+    Args:
+        x (...): Tensor.
+
+    Returns:
+        True if the last two dimensions of `x` are equal, False otherwise.
+    """
+    return x.size(-1) == x.size(-2)
+
+
+def _quantum_type(x: Tensor) -> str:
+    """Get the quantum type of a tensor."""
```

Comment:
Missing `Args`, `Returns` and `Raises` docstring sections.

Reply:
We're also missing a `Raises` section in `ptrace()` if you want to add it.

Reply:
Ho, if we want we can be more permissive with the private function documentation, so you can ignore the first comment.

Reply:
Yes this was not meant to be user-facing so I made a quick doc only. But I can be a bit more thorough if you'd like.